File: 프로그래머스/lv0/120846. 합성수 찾기/합성수 찾기.py
def solution(n):
    # if n < 4: return 0
    # result = n // 2 - 1;
    # for i in range(5, n + 1, 2):
    #     temp = 0;
    #     for j in range(1, i + 1):
    #         if i % j == 0: temp += 1;
    #         if temp == 3:
    #             result += 1;
    #             break;
    # return result
    
    output = 0
    for i in range(4, n + 1):
        for j in range(2, int(i ** 0.5) + 1):
            if i % j == 0:
                output += 1
                break
    return output

File: 프로그래머스/lv0/120846. 합성수 찾기/합성수 찾기.js
function solution(n) {
    // 3이하의 수는 합성수일 수 없다.
    if(n < 4) return 0;
    // 4이상의 모든 짝수는 합성수이다.
    let result = Math.floor(n / 2) - 1;
    // 홀수의 합성수만 검사한다.
    for(let i = 5; i <= n; i += 2){
        let temp = 0;
        for(let j = 1; j <= i; j++){
            if(i % j === 0) temp++;
            if(temp === 3){
                result++;
                break;
            }
        }
    }
    return result
}